src/main/java/com/hierynomus/smbj/session/Session.java

```diff
@@ -186,6 +201,25 @@ public Connection getConnection() {
         }
         return connection.send(packet, signingKeySpec);
     }
+    
+    public <T extends SMB2Packet> T processSendResponse(SMB2CreateRequest packet) throws TransportException {
```

Comment:
It changes the least code for the caller.  If you look at the difference to Share.java, it's not changed by much.